modules/spring-config/src/main/java/org/mule/runtime/config/spring/SpringXmlConfigurationBuilder.java

```diff
@@ -52,72 +42,40 @@
     protected ApplicationContext domainContext;
     protected ApplicationContext parentContext;
     protected ApplicationContext applicationContext;
+    private final ArtifactType artifactType;
 
-    public SpringXmlConfigurationBuilder(String[] configResources, Map<String, String> artifactProperties) throws ConfigurationException
+    public SpringXmlConfigurationBuilder(String[] configResources, Map<String, String> artifactProperties, ArtifactType artifactType) throws ConfigurationException
     {
         super(configResources, artifactProperties);
+        this.artifactType = artifactType;
     }
 
-    public SpringXmlConfigurationBuilder(String configResources, Map<String, String> artifactProperties) throws ConfigurationException
+    public SpringXmlConfigurationBuilder(String configResources, Map<String, String> artifactProperties, ArtifactType artifactType) throws ConfigurationException
     {
-        super(configResources, artifactProperties);
+        this(new String[]{configResources}, artifactProperties, artifactType);
     }
 
-    public SpringXmlConfigurationBuilder(ConfigResource[] configResources, Map<String, String> artifactProperties)
+    public SpringXmlConfigurationBuilder(ConfigResource[] configResources, Map<String, String> artifactProperties, ArtifactType artifactType)
     {
         super(configResources, artifactProperties);
+        this.artifactType = artifactType;
     }
 
     public SpringXmlConfigurationBuilder(String configResource) throws ConfigurationException
     {
-        this(configResource, emptyMap());
+        this(configResource, emptyMap(), APP);
     }
 
     public SpringXmlConfigurationBuilder(String[] configFiles) throws ConfigurationException
     {
         super(configFiles, emptyMap());
-    }
-
-    protected List<ConfigResource> getConfigResources() throws IOException
```

Comment:
Gateway requires overriding this method to add custom stuff of their own. Will they have a similar extension point for them to create custom contexts?